### PR TITLE
[CARBONDATA-1274] Adds example for update and delete with Spark 2.1

### PR DESCRIPTION
**Pull request by mayunSaicmotor**
added detailed examples for update and delete in spark 2.1 and spark1.6
and support examples run on hdfs configuration

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3023/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/435/

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    }
+
+    spark.sql(s"""
+           SELECT count(*) FROM big_table
+           """).show()
+
+    // Drop table
+    spark.sql("DROP TABLE IF EXISTS update_table")
+    spark.sql("DROP TABLE IF EXISTS big_table")
+  }
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: please add spark.stop()  at the end.

**Comment by chenliang613** [comment]
@mayunSaicmotor  Please change PR title to : Adds example for update and delete with Spark 2.1

**Comment by anubhav100** [comment]
@chenliang613 @mayunSaicmotor pr is already raised for this issue https://github.com/apache/carbondata/pull/1158

**Comment by chenliang613** [comment]
@mayunSaicmotor  In the future, to avoid conflict , please assign JIRA to yourself if you plan to work on this PR.
@anubhav100  my side is very appreciating your effort, thanks for your contribution. But mayun is creating this example which would cover all actual use cases.  so can you please close your PR.

**Comment by anubhav100** [comment]
@chenliang613 sure

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      warehouse = "hdfs://nameservice1/carbon2/warehouse/"
+      metastoredb = "hdfs://nameservice1/carbon2/carbonstore/"
+      testData = "hdfs://nameservice1/carbon2/data_update.csv"
+    }
+
+    import org.apache.spark.sql.CarbonSession._
+    val spark = SparkSession
+      .builder()
+      .master("local")
+      .appName("CarbonSessionExample")
```
Comment (anubhav100) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: correct the app name from carbonsession example to DataUpdateDeleteExample

Comment (mayunSaicmotor) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: fixed  thanks

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3054/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/466/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3055/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3057/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/468/

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+      .toDF("ID", "date", "country", "name", "phonetype", "serialname", "salary")
+    df.write
+      .format("carbondata")
+      .option("tableName", "t3")
+      .option("tempCSV", "true")
+      .option("compress", "true")
+      .mode(SaveMode.Overwrite)
+      .save()
+
+    sdf = new SimpleDateFormat("yyyy-MM-dd")
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: Why creates two "sdf" for new SimpleDataFormat, (row 64 and row 77)

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+    df = spark.sparkContext.parallelize(1 to 10)
+      .map(x => (x, new java.sql.Date(sdf.parse("2017-07-" + (x % 20 + 1)).getTime),
+        "usa", "bbb" + x, "phone" + 100 * x, "ASD" + (1000 * x - x), 25000 + x))
+      .toDF("ID", "date", "country", "name", "phonetype", "serialname", "salary")
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: To easier identify data from different table, please change as below : 
toDF("t5.ID", "t5.date", "t5.country", "t5.name", "t5.phonetype", "t5.serialname", "t5.salary")

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    df.write
+      .format("carbondata")
+      .option("tableName", "t5")
+      .option("tempCSV", "true")
+      .option("compress", "true")
+      .mode(SaveMode.Overwrite)
+      .save()
+
+    // Query data again after the above data insert
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: please optimize the comment

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+    // Drop table
+    spark.sql("DROP TABLE IF EXISTS t3")
+    spark.sql("DROP TABLE IF EXISTS t5")
+
+    // use code to create table t5 and insert data
+    var sdf = new SimpleDateFormat("yyyy-MM-dd")
+    var df = spark.sparkContext.parallelize(1 to 10)
+      .map(x => (x, new java.sql.Date(sdf.parse("2015-07-" + (x % 10 + 10)).getTime),
+        "china", "aaa" + x, "phone" + 555 * x, "ASD" + (60000 + x), 14999 + x))
+      .toDF("ID", "date", "country", "name", "phonetype", "serialname", "salary")
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: To easier identify data from different table, please change as below : 
toDF("t3.ID", "t3.date", "t3.country", "t3.name", "t3.phonetype", "t3.serialname", "t3.salary")

File: examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    // Specify date format based on raw data
+    CarbonProperties.getInstance()
+      .addProperty(CarbonCommonConstants.CARBON_DATE_FORMAT, "yyyy-MM-dd")
+
+    import spark.implicits._
+    // Drop table
+    spark.sql("DROP TABLE IF EXISTS t3")
+    spark.sql("DROP TABLE IF EXISTS t5")
+
+    // use code to create table t5 and insert data
```
Comment (chenliang613) [inline examples/spark2/src/main/scala/org/apache/carbondata/examples/DataUpdateDeleteExample.scala]: suggest changing the comment to : // Simulate data and write to table t5

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3071/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/482/

**Comment by chenliang613** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/484/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3073/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/3074/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/485/

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**